dynamic_programming/abbreviation.cpp

```diff
@@ -0,0 +1,192 @@
+/**
+ * @file
+ * @brief Implementation of Abbrievation
+ * (https://www.hackerrank.com/challenges/abbr/problem)
```

Comment:
We use [Markdown](https://guides.github.com/features/mastering-markdown/) format here.
```suggestion
 * @brief [Abbreviation](https://www.hackerrank.com/challenges/abbr/problem) implementation
```